plugin/src/main/java/io/jenkins/plugins/casc/yaml/YamlSource.java

```diff
@@ -77,10 +76,6 @@ public String toString() {
 
     @Override
     public void close() throws IOException {
-        if (reader instanceof BufferedReader) {
-            ((BufferedReader) reader).close();
-        } else if (reader instanceof InputStreamReader) {
-            ((InputStreamReader) reader).close();
-        }
+        read().close();
```

Comment:
Not sure this is correct either.
I am not seeing much benefit in having `YamlReader<T> reader;` and `YamlSource.of` pattern if it comes at the cost of properly closing sources.